### PR TITLE
Render rich-text content from Draftail in Storefront 1.0

### LINKED ISSUE #4300: Render rich-text content from Draftail in Storefront 1.0
Implement support for rendering rich-text content that was created with Draftail editor and it's stored as JSON in Storefront 1.0. There is a library `draftjs-to-html` that we can use for transforming JSON to HTML. We're using it in Storefront 2.0 ([here is the related PR](https://github.com/mirumee/saleor-storefront/pull/361/) that adds it).

### PR DESCRIPTION
**Pull request by AlicjaSzu**
I want to merge this change because...

resolves #4300 

Definition of done:
- [x] The HTML content fields is migrated
- [x] The translations are migrated
- [ ] ~~The dashboard 1.0 editor actually saves into JSON and not in HTML~~
- [x] Add proper handling for:
  - [x] SEO
  - [x] Feeds
  - [x] Avatax
- [x] Users cannot break/inject through JSON
- [x] Plain texts are sanitized 
- [ ] Search inside the fields
- [x] Fix invalid page JSON from populatedb
- [x] The storefront 1.0 renders the JSON

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] Database migration files are up to date.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### CODE REVIEW

**Comment by django-queries** [comment]
Here is the report for 60f2b890e747766e53a047da016a660af9e52f28 (mirumee/saleor @ feature/rich-text-content-strorefront1)
Base comparison is a592e135922054ed6b4cdfad1785d87580613e16.



<details><summary>No differences were found. (click me)</summary>
<p>

```diff
# api.benchmark checkout
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  add billing address to checkout     	         41	         41	            UNK
  add shipping to checkout            	          7	          7	            UNK
  checkout payment charge             	         16	         16	            UNK
  complete checkout                   	          6	          6	            UNK
  create checkout                     	         45	         45	            UNK

# api.benchmark homepage
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve main menu                  	          5	          5	            UNK
  retrieve product list               	          4	          4	            UNK
  retrieve secondary menu             	          5	          5	            UNK
  retrieve shop                       	          2	          2	            UNK

# api.benchmark product
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  product details                     	         13	         13	            UNK

# api.benchmark variant
  test name                           	left count 	right count	duplicate count
  ------------------------------------	-----------	-----------	---------------
  retrieve variant list               	          9	          9	            UNK
```

</p>
</details>

File: saleor/product/views.py [hunk 1]
```diff
@@ -89,7 +89,9 @@ def product_details(request, slug, product_id, form=None):
     # show_variant_picker determines if variant picker is used or select input
     show_variant_picker = all([v.attributes for v in product.variants.all()])
     json_ld_data = product_json_ld(product, product_attributes)
+    description_json = product.translated.description_json
     ctx = {
+        "description_json": json.dumps(description_json, default=serialize_decimal),
```
Comment (NyanKiyoshi) [inline saleor/product/views.py]:
```suggestion
        "description_json": json.dumps(description_json),
```

File: saleor/page/views.py [hunk 2]
```diff
@@ -11,5 +12,6 @@ def page_details(request, slug):
     today = datetime.date.today()
     is_visible = page.publication_date is None or page.publication_date <= today
     return TemplateResponse(
-        request, "page/details.html", {"page": page, "is_visible": is_visible}
+        request, "page/details.html", {"page": page, "is_visible": is_visible,
```
Comment (NyanKiyoshi) [inline saleor/page/views.py]:
```python
    return TemplateResponse(
        request,
        "page/details.html",
        {
            "page": page,
            "is_visible": is_visible,
            "content_json": json.dumps(page.translated.content_json),
        },
    )
```

**Comment by NyanKiyoshi** [comment]
Did we already migrate the old descriptions? We are missing the migrations it looks like (https://github.com/mirumee/saleor/issues/4300#issuecomment-502649676).

Also, I noticed that it might break the SEO bindings (if they are not already broken).

**Comment by NyanKiyoshi** [comment]
Blocked while we work on HTML to DraftJS for exporting old dashboard contents to the new one.

**Comment by maarcingebala** [comment]
I think we should wait with dropping the "old" HTML fields yet. I found that they're used in a few more places:
- `product_json_ld` function
- Google feed
- Avatax module

I see a few problems with it:
- `product_json_ld` should actually use `seo_description` if I'm correct.
- Does the Google feed/Avatax work with descriptions in HTML format? Will they work if we swap it with JSON? I would suppose that they expect plain text, but we have to check it.
- I'm not sure which of the above should come in the translated version (through our model translation proxy).

File: saleor/page/migrations/0008_raw_html_to_json.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+from django.db import migrations
+from html_to_draftjs import html_to_draftjs
+
+
+def convert_pages_html_to_json(apps, schema_editor):
+    Page = apps.get_model("page", "Page")
+    qs = Page.objects.filter(content_json={})
+
+    for page in qs:
+        page.content_json = html_to_draftjs(page.content)
```
Comment (maarcingebala) [inline saleor/page/migrations/0008_raw_html_to_json.py]: Maybe this should happen only for fields that have empty `_json` data? The thing is that even if don't have any content, DraftJS could've already created an empty structure like:
```
{'blocks': [{'key': 'as2m7',
   'data': {},
   'text': 'Test',
   'type': 'unstyled',
   'depth': 0,
   'entityRanges': [],
   'inlineStyleRanges': []}],
 'entityMap': {}}
```
So simple comparison with `None` or empty string wouldn't work.

Comment (NyanKiyoshi) [reply saleor/page/migrations/0008_raw_html_to_json.py]: You mean seeing if the plain text is empty then?

File: saleor/page/migrations/0008_raw_html_to_json.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+def convert_pages_html_to_json(apps, schema_editor):
+    Page = apps.get_model("page", "Page")
+    qs = Page.objects.all()
+
+    for page in qs:
+        content_json = page.content_json
+        content_raw = json_content_to_raw_text(content_json)
+
+        # Override the the JSON content if there was nothing in it
```
Comment (maarcingebala) [inline saleor/page/migrations/0008_raw_html_to_json.py]:
```suggestion
        # Override the JSON content if there was nothing in it
```

File: saleor/page/migrations/0008_raw_html_to_json.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            page.save(update_fields=["content_json"])
+
+    PageTranslation = apps.get_model("page", "PageTranslation")
+    qs = PageTranslation.objects.all()
+
+    for translation in qs:
+        content_json = translation.content_json
+        content_raw = json_content_to_raw_text(content_json)
+
+        # Override the the JSON content if there was nothing in it
```
Comment (maarcingebala) [inline saleor/page/migrations/0008_raw_html_to_json.py]:
```suggestion
        # Override the JSON content if there was nothing in it
```

File: saleor/product/migrations/0096_raw_html_to_json.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+def convert_products_html_to_json(apps, schema_editor):
+    Product = apps.get_model("product", "Product")
+    qs = Product.objects.all()
+
+    for product in qs:
+        description_json = product.description_json
+        description_raw = json_content_to_raw_text(description_json)
+
+        # Override the the JSON description if there was nothing in it
```
Comment (maarcingebala) [inline saleor/product/migrations/0096_raw_html_to_json.py]:
```suggestion
        # Override the JSON description if there was nothing in it
```

File: saleor/product/migrations/0096_raw_html_to_json.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+            product.save(update_fields=["description_json"])
+
+    ProductTranslation = apps.get_model("product", "ProductTranslation")
+    qs = ProductTranslation.objects.all()
+
+    for translation in qs:
+        description_json = translation.description_json
+        description_raw = json_content_to_raw_text(description_json)
+
+        # Override the the JSON description if there was nothing in it
```
Comment (maarcingebala) [inline saleor/product/migrations/0096_raw_html_to_json.py]:
```suggestion
        # Override the JSON description if there was nothing in it
```

**Pull request merged**